docs/flatpak-devel.rst

```diff
@@ -0,0 +1,395 @@
+Flatpak as a developer platform
+===============================
+
+This sections contains an overview on how Flatpak can be used as part
+of a developer workflow.
+
+CI Integration
+--------------
+
+Flatpak can be integrated with the CI pipelines and bundles can be
+produced as artifacts for quick testing or it can be exported to a
+Flatpak repository on a webserver to provide a Nightly Flatpak package.
+
+The CI setup will vary based on the code host and the CI service being
+used. A basic workflow for exporting to Gitlab pages is provided in
+:doc:`hosting-a-repository` which can be used as a starting point.
+
+`GNOME <https://gitlab.gnome.org/GNOME/Initiatives/-/wikis/DevOps-with-Flatpak#basic-ci>`_
+and `KDE <https://develop.kde.org/docs/packaging/flatpak/publishing/#publishing-to-kdes-nightly-repositories>`_
+also has their own respective CI setup for doing this, which can be
+also be used for inspiration.
+
+`Flatpak Github Actions <https://github.com/flathub-infra/flatpak-github-actions>`_
+can be used for GitHub.
+
+Running tests
+-------------
+
+Flatpak Builder can also run tests with ``run-tests, test-rule, test-commands, test-args``.
+
+``run-tests: true`` in a module will run tests for that module after
+installing.
+
+``test-rule`` is the default target to build when running tests. For
+``meson`` an ``cmake-ninja`` buildsystems it defaults to ``test``,
+otherwise ``check`` for ``make``. Setting it to an empty string like
```

Comment:
```suggestion
``meson`` and ``cmake-ninja`` buildsystems it defaults to ``test``,
otherwise ``check`` for ``cmake`` and ``autotools``. Setting it to an empty string like
```

Reply:
thanks